packages/agents/cartographer/poller/src/lib/operations/transfers.ts

```diff
@@ -31,58 +49,122 @@ export const updateTransfers = async () => {
     }
 
     // Retrieve the most recent origin transfers we've saved for this domain.
-    const xCallTimestamp = await database.getLatestXCallTimestamp(domain);
+    const latestOriginNonce = await database.getCheckPoint("origin_nonce_" + domain);
 
-    subgraphXCallQueryMetaParams.set(domain, {
+    subgraphOriginQueryMetaParams.set(domain, {
       maxBlockNumber: latestBlockNumber,
-      fromTimestamp: xCallTimestamp,
+      latestNonce: latestOriginNonce == 0 ? latestOriginNonce : latestOriginNonce + 1,
       orderDirection: "asc",
     });
 
-    const executedTimestamp = await database.getLatestExecuteTimestamp(domain);
+    // Retrieve the most recent destination transfers we've saved for this domain.
+    const latestDestinationNonce = await database.getCheckPoint("destination_nonce_" + domain);
 
-    subgraphExecuteQueryMetaParams.set(domain, {
+    subgraphDestinationQueryMetaParams.set(domain, {
       maxBlockNumber: latestBlockNumber,
-      fromTimestamp: executedTimestamp,
+      latestNonce: latestDestinationNonce == 0 ? latestDestinationNonce : latestDestinationNonce + 1,
       orderDirection: "asc",
     });
 
-    const reconciledTimestamp = await database.getLatestReconcileTimestamp(domain);
+    const reconciledTimestamp = await database.getCheckPoint("destination_reconcile_timestamp_" + domain);
 
     subgraphReconcileQueryMetaParams.set(domain, {
       maxBlockNumber: latestBlockNumber,
       fromTimestamp: reconciledTimestamp,
+      destinationDomains: domains,
       orderDirection: "asc",
     });
+
+    const pendingOriginTransfersIDs = await database.getTransfersWithOriginPending(domain, 100, "ASC");
+
+    subgraphOriginPendingQueryMetaParams.set(domain, {
+      maxBlockNumber: latestBlockNumber,
+      transferIDs: pendingOriginTransfersIDs,
+    });
+
+    const pendingDestinationTransfersIDs = await database.getTransfersWithDestinationPending(domain, 100, "ASC");
+
+    subgraphDestinationPendingQueryMetaParams.set(domain, {
+      maxBlockNumber: latestBlockNumber,
+      transferIDs: pendingDestinationTransfersIDs,
+    });
   }
 
-  if (subgraphXCallQueryMetaParams.size > 0) {
+  if (subgraphOriginQueryMetaParams.size > 0) {
     // Get origin transfers for all domains in the mapping.
-    const transfers = await subgraph.getOriginTransfersByXCallTimestamp(subgraphXCallQueryMetaParams);
-    logger.info("Retrieved origin transfers by xcalled timestamp", requestContext, methodContext, {
+    const transfers = await subgraph.getOriginTransfersByNonce(subgraphOriginQueryMetaParams);
+    logger.info("Retrieved origin transfers by nonce", requestContext, methodContext, {
       transfers,
       count: transfers.length,
     });
     await database.saveTransfers(transfers);
+
+    for (const domain of domains) {
```

Comment:
Nit, but this should be able to be parallelized i.e.

```ts
await Promise.all(domains.map(async (domain) => {
  ...
});
```

Reply:
Yes!